---
title: "MACseq Plate 2 Analysis APR2025"
output: html_document
date: "2025-04-28"
---

```{r setup}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, fig.height=8, fig.width = 10)
```

```{r}
### loading required packages
library(tidyverse)
library(Seurat)
library(gtools)
library(tidyseurat)
library(DESeq2)
library(edgeR)
library(DT)
library(dplyr)
library(pheatmap)
library(EnhancedVolcano)
library(RColorBrewer)
library(ggvenn)
library(gplots)
library(clusterProfiler)
library(enrichplot)
```

```{r}
### Functions 

source('/Users/dalvinikita/Desktop/macpie/R/filter_genes_by_expression.R')

### buttons for datatables 
create_dt_genes <- function(x){
  colnames <- c('logFC','AveExpr','t','adj.P.Val')
  df <- x %>%
    mutate(across(all_of(colnames),round, digits = 2)) %>%
    mutate(adj.P.Val = format(adj.P.Val, format='e', digits=4)) %>%
    mutate(P.Value = format(P.Value, format='e', digits=4))
  DT::datatable(df,
                extensions = 'Buttons',
                options = list(dom = 'Blfrtip',
                               buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                               lengthMenu = list(c(10,25,50,-1),
                                                 c(10,25,50,"All"))))
}

create_dt <-  function(x){
  datatable(x,
            extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
}


plotRLE <- function(data, main = "", ylab = "Relative Log Expression") {
  # Calculate medians
  col <- brewer.pal(8, 'Set3')
  medians <- apply(data, 1, median, na.rm = TRUE)
  rle_data <- data - medians
  
  boxplot(as.data.frame(rle_data),
          outline = FALSE,
          las = 2,
          ylab = ylab,
          main = main,
          col = col)
  abline(h = 0, col = "red", lty = 2)
}

red_blue_palette <- colorRampPalette(brewer.pal(11, "RdBu"))(100)
```

## Notes

Thresholds used for DE analysis:

• pvalue threshold = 0.05

• log FC threshold = |lfc| > 0.58, which is equivalent to an absolute FC of |fc| > 1.5

```{r, echo=TRUE}
### Threshold Constants

absolute_fold_change <- 1.0

p_value_threshold <- 0.05
```

## Data

```{r}
### Importing MACseq data
### Importing metadata
### Joining data in a Seurat object 

### import metadata
metadata<-read.csv("/Users/dalvinikita/Documents/11_MACseq_PMMSq042/PMMSq042.csv",header=T) %>%
  arrange("Barcode") %>%
  mutate(id=gsub("Plate","",Plate_ID)) %>%
  mutate(Treatment_Combination=paste0(substr(Cell_type,1,2),'_',substr(Treatment_1,1,4))) %>%
  mutate(label=paste0(Treatment_Combination,"_",Well_ID)) %>%
  mutate(Concentration_1=as.factor(Concentration_1)) %>%
  mutate(SPLINTR = str_sub(Cell_type,-1,-1)) %>%
  mutate(group = paste0(Treatment_Combination,'.',SPLINTR)) %>%
  mutate(combined_id = paste0(Cell_type, '_', Treatment_1))


### load in count data 
raw_counts <- Read10X(data.dir = '/Users/dalvinikita/Documents/11_MACseq_PMMSq042/raw')

### filter for the minimum number of reads per gene per sample, as per edgeR
row.names(metadata)<-metadata$label

mac <- CreateSeuratObject(counts = raw_counts,
                          project = 'PMMSq042',
                          min.cells = 1,
                          min.features = 1)

mac[["percent.mt"]] <- PercentageFeatureSet(mac, pattern = "^mt-|^MT-")
mac[["percent.ribo"]] <- PercentageFeatureSet(mac, pattern = "^Rp[slp][[:digit:]]|^Rpsa|^RP[SLP][[:digit:]]|^RPSA")

### add location metadata
mac<- mac %>%
  inner_join(metadata,by=c(".cell"="Barcode"))

mac <- filter_genes_by_expression(mac,
                                       group_by = "Organoid",
                                       min_counts = 10,
                                       min_samples = 80)

### exclude mito and ribo genes from the counts
mac<-mac[!grepl("^MT-",rownames(mac))]
mac<-mac[!grepl("^RP[SL][[:digit:]]|^RPLP[[:digit:]]|^RPSA",rownames(mac))]

colnames(mac)<- mac$label
```

A quick look at the metadata:

```{r}
create_dt(metadata[metadata$Sample_type!="EMPTY",c("Well_ID","Barcode","Organoid", "Treatment_Combination", "SPLINTR", "Treatment_1")])
```

## Basic Visualisation {.tabset}

### Violin Plot

```{r, fig.height=5, fig.width=8}
VlnPlot(mac, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.ribo"),
    ncol = 4, group.by = "Organoid")

VlnPlot(mac, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.ribo"),
    ncol = 4, group.by = "Sample_type")
```

### Pre and post normalisation {.tabset}

#### Entire Plate

DMSO raw VS normalised wells across entire plate:

```{r, fig.width=18, fig.height=6}
###   Select wells from mac that are not empty 
rawdata <- mac
sampleinfo <- metadata[is.na(metadata$Cell_type)!=TRUE | metadata$Treatment_1!="Staurosporine",]
samples <- rownames(sampleinfo)
rawdata <- mac[,samples]

### Keep only DMSO treated wells
rawdata <- rawdata %>%
  filter(Treatment_1=="DMSO")

### Ensure counts data and sampleinfo are in the same order
order <- colnames(rawdata)
sampleinfo <- sampleinfo[order,]
order_check <- rownames(sampleinfo)==colnames(rawdata)
cat('Instances where metadata sample matches counts matric sample (checking correct order):',"\n", order_check)

dge <- DGEList(counts = rawdata@assays$RNA$counts)
dge <- calcNormFactors(dge, methods = "TMMwsp")
treatment_combination <- as.factor(sampleinfo$Treatment_Combination)
splintr <- as.factor(sampleinfo$SPLINTR)
concentration <- as.factor(sampleinfo$Concentration_1)
organoid <- as.factor(sampleinfo$Organoid)
design <- model.matrix(~0+organoid+treatment_combination+splintr+concentration)
v <- voom(dge, design)

par(mfrow=c(1,2))
plotRLE(as.data.frame(rawdata@assays$RNA$counts), main='RLE of all DMSO treated wells - raw')
plotRLE(v$E, main='RLE of all DMSO treated wells - normalised')

```

## Analysis

### ORG38 

#### Design Matrix and Contrasts 

Factors to include in design matrix: 

• SPLINTR/treatment group (splintrA/ splintrB)

• Treatment combination

• Concentration (High/Low for each treatment combination)

Design matrix:

```{r}
### selecting metadata and countdata relevant for ORG 38
### subset metadata to include only ORG38 cells 
columns <- metadata[metadata$Organoid=='ORG38' & metadata$Treatment_1!="Staurosporine",]
sampleinfo <- columns
samples <- c(rownames(sampleinfo))
counts_matrix <- mac[,samples]
order <- colnames(counts_matrix)
sampleinfo <- sampleinfo[colnames(counts_matrix),]

### factors to include in deisgn matrix: SPLINTR group and treatment, concentration
treatment <- as.factor(sampleinfo$Treatment_Combination)
concentration <- as.factor(sampleinfo$Concentration_1)
splintr <- as.factor(sampleinfo$SPLINTR)

### set up design matrix
### without intercept (means model)
design <- model.matrix(~0+treatment+splintr+concentration)
rownames(design) <- rownames(sampleinfo)
create_dt(design)
org38_design <- design

NC_pacvsdmso <- makeContrasts(treatmentNC_Pacl-treatmentNC_DMSO,levels=colnames(design))
NC_sn38vsdmso <- makeContrasts(treatmentNC_SN38-treatmentNC_DMSO,levels=colnames(design))
PAC_pacvsdmso <- makeContrasts(treatmentPa_Pacl-treatmentPa_DMSO,levels=colnames(design))
SN38_sn38vsdmso <- makeContrasts(treatmentSN_SN38-treatmentSN_DMSO,levels=colnames(design))
long_PAC <- makeContrasts((treatmentPa_Pacl-treatmentPa_DMSO)-(treatmentNC_Pacl-treatmentNC_DMSO),levels=colnames(design))
long_SN38 <- makeContrasts((treatmentSN_SN38-treatmentSN_DMSO)-(treatmentNC_SN38-treatmentNC_DMSO),levels=colnames(design))

```

#### Differential Expression {.tabset}

Voom plot: 

```{r, fig.height=5, fig.width=5}
### Normalising data
counts<-counts_matrix@assays$RNA$counts
counts <- DGEList(counts = counts) %>%
  calcNormFactors(method = 'TMMwsp')
v <- voom(counts, design, plot = T)
org38_voom_mat <- v
```

RLE of all ORG38 wells before and after normalisation:

```{r, fig.width=16}
par(mfrow=c(1,2))
plotRLE(counts$counts, main='RLE of all wells - raw')
plotRLE(v$E, main='RLE of all wells - normalised')
```

##### NC_PAC-NC_DMSO

Differentially expressed genes between NC Pac and NC DMSO in ORG38:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org38_ncPAC_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org38_ncPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_ncPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_ncPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_ncPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_ncPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

```{r, fig.height=6, fig.width=8}
### Assign Significance 
all_genes$significance <- 'nonsignificant'
all_genes[all_genes$logFC < -1 & all_genes$adj.P.Val < 0.05,c('significance')] <- 'downregulated'
all_genes[all_genes$logFC >1 & all_genes$adj.P.Val < 0.05,c('significance')] <- 'upregulated'

### MA plot
MA <- new("MAList")
MA$A <- all_genes$AveExpr
MA$M <- all_genes$logFC
status <- all_genes$significance
values <- c("downregulated","upregulated","nonsignificant")
col <- c("blue", "red", "black")
plotMA(MA,main="MA-Plot with significant genes highlighted",
       status=status, values=values, hl.col=col)
```


Next, we can look the differentially expressed genes by visualising the voom-transformed counts of the treated and DMSO samples in a heatmap, to make sure that the DEgenes we've identified actually look "different" in terms of their expression. 

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_ncPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_Pacl'|sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)
```

Heatmap of top 50 differentially expressed genes between NC_PAC and NC_DMSO in PAC/DMSO samples in ORG38:

```{r}
pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```


Volcano plot of differentially expressed genes with top 10 differentially expressed genes labelled:

```{r, fig.height=6, fig.width=6}
### Assigning different colours for upregulated/downregulates genes
keyvals <- ifelse(
    all_genes$logFC < -1, 'darkred',
      ifelse(all_genes$logFC > 1, 'darkgreen',
        'black'))
  keyvals[is.na(keyvals)] <- 'black'
  names(keyvals)[keyvals == 'darkgreen'] <- 'UP'
  names(keyvals)[keyvals == 'black'] <- 'NO'
  names(keyvals)[keyvals == 'darkred'] <- 'DOWN'
  
### Volcano plot 
volcano_plot <- EnhancedVolcano(all_genes,
                                title="NC Paclitaxel VS NC DMSO",
                                subtitle="ORG38",
                                lab=rownames(all_genes),
                                selectLab = rownames(org38_ncPAC_DEgenes)[1:10],
                                boxedLabels = TRUE,
                                drawConnectors = TRUE,
                                arrowheads = FALSE,
                                pointSize = 3,
                                colAlpha = 1, 
                                x='logFC', 
                                y='adj.P.Val', 
                                pCutoff=0.05,
                                FCcutoff = 1,
                                border="full",
                                legendPosition = "bottom",
                                colCustom = keyvals)
volcano_plot
```

```{r}
### returning a list of corresponding entrez IDs for gene names in DE genes
entrezids <- bitr(rownames(org38_ncPAC_DEgenes), fromType=c("SYMBOL"), toType="ENTREZID", OrgDb="org.Hs.eg.db")

### add entrezid data
org38_ncPAC_DEgenes$gene <- rownames(org38_ncPAC_DEgenes)
pathway_data<- left_join(org38_ncPAC_DEgenes, entrezids, by=join_by(gene==SYMBOL))

### removing genes that dont map to an entrezID (~0%)
pathway_data <- drop_na(pathway_data)

### setting up gene list
geneList <- pathway_data$logFC
names(geneList)<- pathway_data$ENTREZID

#performing reactome enrichment analysis 
x <- ReactomePA::enrichPathway(gene=names(geneList),
                               organism='human',
                               pvalueCutoff = 1,
                               qvalueCutoff = 1,
                               readable=TRUE)
x2 <- pairwise_termsim(x)
enrichplot::treeplot(x2, fontsize=2, nWords=3)
```


##### NC_SN38-NC_DMSO

Differentially expressed genes between NC SN38 and NC DMSO in ORG38:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org38_ncSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org38_ncSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_ncSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_ncSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_ncSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_ncSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

```{r}
### Assign Significance 
all_genes$significance <- 'nonsignificant'
all_genes[all_genes$logFC < -1 & all_genes$adj.P.Val < 0.05,c('significance')] <- 'downregulated'
all_genes[all_genes$logFC >1 & all_genes$adj.P.Val < 0.05,c('significance')] <- 'upregulated'

### MA plot
MA <- new("MAList")
MA$A <- all_genes$AveExpr
MA$M <- all_genes$logFC
status <- all_genes$significance
values <- c("downregulated","upregulated","nonsignificant")
col <- c("blue", "red", "black")
plotMA(MA,main="MA-Plot with significant genes highlighted",
       status=status, values=values, hl.col=col)
```

Heatmap of top 50 differentially expressed genes between NC_SN38 and NC_DMSO in SN38/DMSO samples in ORG38:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_ncSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_SN38'|sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

```{r}
### returning a list of corresponding entrez IDs for gene names in DE genes
entrezids <- bitr(rownames(org38_ncSN38_DEgenes), fromType=c("SYMBOL"), toType="ENTREZID", OrgDb="org.Hs.eg.db")

### add entrezid data
org38_ncSN38_DEgenes$gene <- rownames(org38_ncSN38_DEgenes)
pathway_data<- left_join(org38_ncSN38_DEgenes, entrezids, by=join_by(gene==SYMBOL))

### removing genes that dont map to an entrezID (~0%)
pathway_data <- drop_na(pathway_data)

### setting up gene list
geneList <- pathway_data$logFC
names(geneList)<- pathway_data$ENTREZID

#performing reactome enrichment analysis 
x <- ReactomePA::enrichPathway(gene=names(geneList),
                               organism='human',
                               pvalueCutoff = 1,
                               qvalueCutoff = 1,
                               readable=TRUE)
x2 <- pairwise_termsim(x)
enrichplot::treeplot(x2, fontsize=2, nWords=3)
```

##### PAC_PAC - PAC_DMSO

Differentially expressed genes between PAC_PAC and PAC_DMSO in ORG38:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, PAC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org38_paPAC_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org38_paPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_paPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_paPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_paPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_paPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of differentially expressed genes between PAC_PAC nad PAC_DMSO in ORG38:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_paPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='Pa_Pacl'|sampleinfo$Treatment_Combination=='Pa_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

```{r}
### returning a list of corresponding entrez IDs for gene names in DE genes
entrezids <- bitr(rownames(org38_paPAC_DEgenes), fromType=c("SYMBOL"), toType="ENTREZID", OrgDb="org.Hs.eg.db")

### add entrezid data
org38_paPAC_DEgenes$gene <- rownames(org38_paPAC_DEgenes)
pathway_data<- left_join(org38_paPAC_DEgenes, entrezids, by=join_by(gene==SYMBOL))

### removing genes that dont map to an entrezID (~0%)
pathway_data <- drop_na(pathway_data)

### setting up gene list
geneList <- pathway_data$logFC
names(geneList)<- pathway_data$ENTREZID

#performing reactome enrichment analysis 
x <- ReactomePA::enrichPathway(gene=names(geneList),
                               organism='human',
                               pvalueCutoff = 1,
                               qvalueCutoff = 1,
                               readable=TRUE)
x2 <- pairwise_termsim(x)
enrichplot::treeplot(x2, fontsize=2, nWords=3)
```

##### SN38_SN38 - SN38_DMSO)

Differentially expressed genes between SN38_SN38 and SN38_DMSO in ORG38:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, SN38_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org38_snSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org38_snSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_snSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_snSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_snSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_snSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of top 50 differentially expressed genes between SN_SN38 and SN_DMSO in SN38/DMSO samples in ORG38:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_snSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='SN_SN38'|sampleinfo$Treatment_Combination=='SN_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(10)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

```{r}
### returning a list of corresponding entrez IDs for gene names in DE genes
entrezids <- bitr(rownames(org38_snSN38_DEgenes), fromType=c("SYMBOL"), toType="ENTREZID", OrgDb="org.Hs.eg.db")

### add entrezid data
org38_snSN38_DEgenes$gene <- rownames(org38_snSN38_DEgenes)
pathway_data<- left_join(org38_snSN38_DEgenes, entrezids, by=join_by(gene==SYMBOL))

### removing genes that dont map to an entrezID (~0%)
pathway_data <- drop_na(pathway_data)

### setting up gene list
geneList <- pathway_data$logFC
names(geneList)<- pathway_data$ENTREZID

#performing reactome enrichment analysis 
x <- ReactomePA::enrichPathway(gene=names(geneList),
                               organism='human',
                               pvalueCutoff = 1,
                               qvalueCutoff = 1,
                               readable=TRUE)
x2 <- pairwise_termsim(x)
enrichplot::treeplot(x2, fontsize=2, nWords=3)
```

##### (PAC_PAC - PAC_DMSO) - (NC_PAC-NC_DMSO)

Differentially expressed genes between (PAC_PAC - PAC_DMSO) and (NC_PAC-NC_DMSO) in ORG38:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_PAC)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org38_longPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_longPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_longPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_longPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_longPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of differentially expressed genes between (PAC_PAC - PAC_DMSO) and (NC_PAC-NC_DMSO) in PAC/DMSO samples in ORG38:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_longPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='Pa_Pacl'|
                                    sampleinfo$Treatment_Combination=='Pa_DMSO'|
                                    sampleinfo$Treatment_Combination=='NC_DMSO'|
                                    sampleinfo$Treatment_Combination=='NC_Pacl',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Initial_Treatment = as.factor(sampleinfo[,c(10)]),
                              Treatment_Combination = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes,],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 20,
         color = red_blue_palette)
```


##### (SN38_SN38 - SN38_DMSO) - (NC_SN38-NC_DMSO)

Differentially expressed genes between (SN38_SN38 - SN38_DMSO) and (NC_SN38-NC_DMSO) in ORG38:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_SN38)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org38_longSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org38_longSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org38_longSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org38_longSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org38_longSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of differentially expressed genes between (SN38_SN38 - SN38_DMSO) and (NC_SN38-NC_DMSO) in SN38/DMSO samples in ORG38:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org38_longSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='SN_SN38'|
                                    sampleinfo$Treatment_Combination=='SN_DMSO'|
                                    sampleinfo$Treatment_Combination=='NC_DMSO'|
                                    sampleinfo$Treatment_Combination=='NC_SN38',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Initial_Treatment = as.factor(sampleinfo[,c(10)]),
                              Treatment_Combination = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes,],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 20,
         color = red_blue_palette)
```

#### ORG38 Gene list comparison {.tabset}

```{r}
### Venn Diagram tests

x <- list(`nc(PAC-DMSO)`=rownames(org38_ncPAC_DEgenes),
          `nc(SN38-DMSO)`=rownames(org38_ncSN38_DEgenes),
          `pa(PAC-DMSO)`=rownames(org38_paPAC_DEgenes),
          `sn(SN38-DMSO)`=rownames(org38_snSN38_DEgenes))

```

##### ncPAC VS ncSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(PAC-DMSO)","nc(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(PAC-DMSO)','nc(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(PAC-DMSO) and nc(SN38-DMSO) : 

  • `r common_genes`

##### ncPAC VS paPAC
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(PAC-DMSO)","pa(PAC-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(PAC-DMSO)','pa(PAC-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):pa(PAC-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(PAC-DMSO) and pa(PAC-DMSO) : 

  • `r common_genes`

##### ncSN38 VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(SN38-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(SN38-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(SN38-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(SN38-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### paPAC VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("pa(PAC-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('pa(PAC-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between pa(PAC-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### all comparisons
```{r}
x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO):pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between all treatments:

  • `r common_genes`

### ORG49 

#### Design Matrix and Contrasts 

Factors to include in design matrix: 

• SPLINTR/treatment group (splintrA/ splintrB)

• Treatment combination

• Concentration

Design matrix:

```{r}
### selecting metadata and countdata relevant for ORG 49
### subset metadata to include only ORG49 cells 
columns <- metadata[metadata$Organoid=='ORG49' & metadata$Treatment_1!="Staurosporine",]
sampleinfo <- columns
samples <- c(rownames(sampleinfo))
counts_matrix <- mac[,samples]
order <- colnames(counts_matrix)
sampleinfo <- sampleinfo[colnames(counts_matrix),]

### factors to include in deisgn matrix: SPLINTR group and treatment, concentration
treatment <- as.factor(sampleinfo$Treatment_Combination)
concentration <- as.factor(sampleinfo$Concentration_1)
splintr <- as.factor(sampleinfo$SPLINTR)

### set up design matrix
### without intercept (means model)
design <- model.matrix(~0+treatment+splintr+concentration)
rownames(design) <- rownames(sampleinfo)
create_dt(design)
org49_design <- design

NC_pacvsdmso <- makeContrasts(treatmentNC_Pacl-treatmentNC_DMSO,levels=colnames(design))
NC_sn38vsdmso <- makeContrasts(treatmentNC_SN38-treatmentNC_DMSO,levels=colnames(design))
PAC_pacvsdmso <- makeContrasts(treatmentPa_Pacl-treatmentPa_DMSO,levels=colnames(design))
SN38_sn38vsdmso <- makeContrasts(treatmentSN_SN38-treatmentSN_DMSO,levels=colnames(design))
long_PAC <- makeContrasts((treatmentPa_Pacl-treatmentPa_DMSO)-(treatmentNC_Pacl-treatmentNC_DMSO),levels=colnames(design))
long_SN38 <- makeContrasts((treatmentSN_SN38-treatmentSN_DMSO)-(treatmentNC_SN38-treatmentNC_DMSO),levels=colnames(design))

```

#### Differential Expression {.tabset}

Voom plot: 

```{r, fig.height=5, fig.width=5}
### Normalising data
counts<-counts_matrix@assays$RNA$counts
counts <- DGEList(counts = counts) %>%
  calcNormFactors(method = 'TMMwsp')
v <- voom(counts, design, plot = T)
org49_voom_mat <- v
```

RLE of all ORG49 wells before and after normalisation:

```{r, fig.width=16}
par(mfrow=c(1,2))
plotRLE(counts$counts, main='RLE of all wells - raw')
plotRLE(v$E, main='RLE of all wells - normalised')
```

##### NC_PAC-NC_DMSO

Differentially expressed genes between NC Pac and NC DMSO in ORG49:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org49_ncPAC_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org49_ncPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_ncPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_ncPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_ncPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

##### NC_SN38-NC_DMSO

Differentially expressed genes between NC SN38 and NC DMSO in ORG49:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org49_ncSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org49_ncSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org49_ncSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_ncSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_ncSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_ncSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of differentially expressed genes between NC_SN38 and NC_DMSO in SN38/DMSO samples in ORG49:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org49_ncSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_SN38'|sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes,],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### PAC_PAC - PAC_DMSO

Differentially expressed genes between PAC_PAC and PAC_DMSO in ORG49:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, PAC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org49_paPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_paPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_paPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_paPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

##### SN38_SN38 - SN38_DMSO)

Differentially expressed genes between SN38_SN38 and SN38_DMSO in ORG49:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, SN38_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org49_snSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org49_snSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org49_snSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_snSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_snSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_snSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of top 50 differentially expressed genes between SN_SN38 and SN_DMSO in SN38/DMSO samples in ORG49:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org49_snSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='SN_SN38'|sampleinfo$Treatment_Combination=='SN_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### (PAC_PAC - PAC_DMSO) - (NC_PAC-NC_DMSO)

Differentially expressed genes between (PAC_PAC - PAC_DMSO) and (NC_PAC-NC_DMSO) in ORG49:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_PAC)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org49_longPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_longPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_longPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_longPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

##### (SN38_SN38 - SN38_DMSO) - (NC_SN38-NC_DMSO)

Differentially expressed genes between (SN38_SN38 - SN38_DMSO) and (NC_SN38-NC_DMSO) in ORG49:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_SN38)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org49_longSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org49_longSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org49_longSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org49_longSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org49_longSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of  differentially expressed genes between SN38 and DMSO in SN38/DMSO samples in ORG49:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org49_longSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_SN38'|
                                    sampleinfo$Treatment_Combination=='NC_DMSO'|
                                    sampleinfo$Treatment_Combination=='SN_DMSO'|
                                    sampleinfo$Treatment_Combination=='SN_SN38',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

#### ORG49 Paclitaxel sample testing

##### Heatmaps 

```{r}
### Heatmap of ORG38 nc_pac genes in ORG49
genes <- rownames(org38_ncPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_Pacl'|
                                    sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))

### Heatmap of org38 PAC_PAC genes
genes <- rownames(org38_paPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='Pa_Pacl'|
                                    sampleinfo$Treatment_Combination=='Pa_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))

```

##### Boxplots 

```{r}
v$targets$Treatment_Combination <- counts_matrix$Treatment_Combination

boxplot(counts_matrix@assays$RNA$counts["SCD",] ~ counts_matrix$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of raw RNA counts - SCD')

boxplot(v$E["SCD",] ~ v$targets$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of voom transformed RNA counts - SCD')

boxplot(counts_matrix@assays$RNA$counts["CA9",] ~ counts_matrix$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of raw RNA counts - CA9')

boxplot(v$E["CA9",] ~ v$targets$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of voom transformed RNA counts - CA9')

boxplot(counts_matrix@assays$RNA$counts["AQP5",] ~ counts_matrix$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of raw RNA counts - AQP5')

boxplot(v$E["AQP5",] ~ v$targets$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of voom transformed RNA counts - AQP5')

boxplot(counts_matrix@assays$RNA$counts["PLK1",] ~ counts_matrix$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of raw RNA counts - PLK1')

boxplot(v$E["PLK1",] ~ v$targets$Treatment_Combination, col=c('lightblue','skyblue3','skyblue4','pink','palevioletred','purple1', 'purple4'))
title('boxplot of voom transformed RNA counts - PLK1')

```

#### ORG49 Gene list comparison {.tabset}

```{r}
### Venn Diagram tests

x <- list(`nc(PAC-DMSO)`=rownames(org49_ncPAC_DEgenes),
          `nc(SN38-DMSO)`=rownames(org49_ncSN38_DEgenes),
          `pa(PAC-DMSO)`=rownames(org49_paPAC_DEgenes),
          `sn(SN38-DMSO)`=rownames(org49_snSN38_DEgenes))

```

##### ncPAC VS ncSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(PAC-DMSO)","nc(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(PAC-DMSO)','nc(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(PAC-DMSO) and nc(SN38-DMSO) : 

  • `r common_genes`

##### ncPAC VS paPAC

0 DEGs identified for both comparisons. 


##### ncSN38 VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(SN38-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(SN38-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(SN38-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(SN38-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### paPAC VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("pa(PAC-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('pa(PAC-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between pa(PAC-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### all comparisons
```{r}
x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO):pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between all treatments

### ORG66 

#### Design Matrix and Contrasts 

Factors to include in design matrix: 

• SPLINTR/treatment group (splintrA/ splintrB)

• Treatment combination

• Concentration

Design matrix:

```{r}
### selecting metadata and countdata relevant for ORG 66
### subset metadata to include only ORG66 cells 
columns <- metadata[metadata$Organoid=='ORG66' & metadata$Treatment_1!="Staurosporine",]
sampleinfo <- columns
samples <- c(rownames(sampleinfo))
counts_matrix <- mac[,samples]
order <- colnames(counts_matrix)
sampleinfo <- sampleinfo[colnames(counts_matrix),]

### factors to include in deisgn matrix: SPLINTR group and treatment, concentration
treatment <- as.factor(sampleinfo$Treatment_Combination)
concentration <- as.factor(sampleinfo$Concentration_1)
splintr <- as.factor(sampleinfo$SPLINTR)

### set up design matrix
### without intercept (means model)
design <- model.matrix(~0+treatment+splintr+concentration)
rownames(design) <- rownames(sampleinfo)
datatable(design)
org66_design <- design

NC_pacvsdmso <- makeContrasts(treatmentNC_Pacl-treatmentNC_DMSO,levels=colnames(design))
NC_sn38vsdmso <- makeContrasts(treatmentNC_SN38-treatmentNC_DMSO,levels=colnames(design))
PAC_pacvsdmso <- makeContrasts(treatmentPa_Pacl-treatmentPa_DMSO,levels=colnames(design))
SN38_sn38vsdmso <- makeContrasts(treatmentSN_SN38-treatmentSN_DMSO,levels=colnames(design))
long_PAC <- makeContrasts((treatmentPa_Pacl-treatmentPa_DMSO)-(treatmentNC_Pacl-treatmentNC_DMSO),levels=colnames(design))
long_SN38 <- makeContrasts((treatmentSN_SN38-treatmentSN_DMSO)-(treatmentNC_SN38-treatmentNC_DMSO),levels=colnames(design))
```

#### Differential Expression {.tabset}

Voom plot: 

```{r, fig.height=5, fig.width=5}
### Normalising data
counts<-counts_matrix@assays$RNA$counts
counts <- DGEList(counts = counts) %>%
  calcNormFactors(method = 'TMMwsp')
v <- voom(counts, design, plot = T)
org66_voom_mat <- v
```

RLE of all ORG38 wells before and after normalisation:

```{r, fig.width=16}
par(mfrow=c(1,2))
plotRLE(counts$counts, main='RLE of all wells - raw')
plotRLE(v$E, main='RLE of all wells - normalised')
```

##### NC_PAC-NC_DMSO

Differentially expressed genes between NC Pac and NC DMSO in ORG66:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org66_ncPAC_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org66_ncPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_ncPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_ncPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_ncPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_ncPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of  differentially expressed genes between PAC and DMSO in PAC/DMSO samples in ORG66:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org66_ncPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_Pacl'|
                                    sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]

heatmap_counts <- as.data.frame(v$E[,samples_of_interest])

### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes,],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### NC_SN38-NC_DMSO

Differentially expressed genes between NC SN38 and NC DMSO in ORG66:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, NC_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org66_ncSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org66_ncSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_ncSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_ncSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_ncSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_ncSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of top 50 differentially expressed genes between NC_SN38 and NC_DMSO in SN38/DMSO samples in ORG66:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org66_ncSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='NC_SN38'|sampleinfo$Treatment_Combination=='NC_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### PAC_PAC - PAC_DMSO

Differentially expressed genes between PAC_PAC and PAC_DMSO in ORG66:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, PAC_pacvsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org66_paPAC_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org66_paPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_paPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_paPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_paPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_paPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of  differentially expressed genes between PAC_PAC and PAC_DMSO in PAC/DMSO samples in ORG66:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org66_paPAC_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='Pa_Pacl'|
                                    sampleinfo$Treatment_Combination=='Pa_DMSO',c('label')]

heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### SN38_SN38 - SN38_DMSO)

Differentially expressed genes between SN38_SN38 and SN38_DMSO in ORG66:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, SN38_sn38vsdmso)
fit <- treat(fit, fc=absolute_fold_change)
org66_snSN38_fit <- fit
all_genes <- topTreat(fit, number=Inf)
org66_snSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_snSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_snSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_snSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_snSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

Heatmap of differentially expressed genes between SN_SN38 and SN_DMSO in SN38/DMSO samples in ORG66:

```{r}
### Assigning genes of interest from DEgenes table
### Formatting counts to log(CPM) for heatmap
### subsetting wells of interest
genes <- rownames(org66_snSN38_DEgenes)
samples_of_interest <- sampleinfo[sampleinfo$Treatment_Combination=='SN_SN38'|sampleinfo$Treatment_Combination=='SN_DMSO',c('label')]
heatmap_counts <- as.data.frame(v$E[,samples_of_interest])


### Setting up annotation data frame
type_annotation <- data.frame(Treatment = as.factor(sampleinfo[,c(18)]),
                              Concentration = as.factor(sampleinfo[,c(11)]),
                              SPLINTR = as.factor(sampleinfo[,c(16)]))
rownames(type_annotation) <- rownames(sampleinfo)

pheatmap(heatmap_counts[genes[1:50],],
         border_color = 'NA',
         annotation_col = type_annotation,
         treeheight_row = 0,
         cellheight = 8,
         color=rev(red_blue_palette))
```

##### (PAC_PAC - PAC_DMSO) - (NC_PAC-NC_DMSO)

Differentially expressed genes between (PAC_PAC - PAC_DMSO) and (NC_PAC-NC_DMSO) in ORG66:

```{r}
### following limma workflow
### difference between PAC and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_PAC)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org66_longPAC_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_longPAC_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_longPAC_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_longPAC_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_longPAC_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```


##### (SN38_SN38 - SN38_DMSO) - (NC_SN38-NC_DMSO)

Differentially expressed genes between (SN38_SN38 - SN38_DMSO) and (NC_SN38-NC_DMSO) in ORG66:

```{r}
### following limma workflow
### difference between SN38 and DMSO across concentrations
fit <- lmFit(v, design)
fit <- contrasts.fit(fit, long_SN38)
fit <- treat(fit, fc=absolute_fold_change)
all_genes <- topTreat(fit, number=Inf)
org66_longSN38_DEgenes <- topTreat(fit, p.value = p_value_threshold, number=Inf, sort.by='p')

create_dt_genes(org66_longSN38_DEgenes)

### Print summary of DE genes table
total <- paste('• TOTAL DE genes:',length(rownames(org66_longSN38_DEgenes)))
up <- paste('• UPREGULATED:',sum(org66_longSN38_DEgenes$logFC>0))
down <- paste('• DOWNREGULATED:',sum(org66_longSN38_DEgenes$logFC<0))
summary=paste (total, up, down, sep="\n")
cat(summary[1])
```

#### ORG66 Gene list comparison {.tabset}

```{r}
### Venn Diagram df

x <- list(`nc(PAC-DMSO)`=rownames(org66_ncPAC_DEgenes),
          `nc(SN38-DMSO)`=rownames(org66_ncSN38_DEgenes),
          `pa(PAC-DMSO)`=rownames(org66_paPAC_DEgenes),
          `sn(SN38-DMSO)`=rownames(org66_snSN38_DEgenes))

```

##### ncPAC VS ncSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(PAC-DMSO)","nc(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(PAC-DMSO)','nc(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(PAC-DMSO) and nc(SN38-DMSO) : 

  • `r common_genes`

##### ncPAC VS paPAC
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(PAC-DMSO)","pa(PAC-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(PAC-DMSO)','pa(PAC-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):pa(PAC-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(PAC-DMSO) and pa(PAC-DMSO) : 

  • `r common_genes`

##### ncSN38 VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("nc(SN38-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('nc(SN38-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(SN38-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between nc(SN38-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### paPAC VS snSN38
```{r, fig.height=6, fig.width=6}
x.table <- venn(x[c("pa(PAC-DMSO)","sn(SN38-DMSO)")], show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, columns = c('pa(PAC-DMSO)','sn(SN38-DMSO)'), 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were found to be shared between pa(PAC-DMSO) and sn(SN38-DMSO) : 

  • `r common_genes`

##### all comparisons
```{r}
x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`nc(PAC-DMSO):nc(SN38-DMSO):pa(PAC-DMSO):sn(SN38-DMSO)`
```

`r length(common_genes)` genes were shared between all treatments

### Comparing ORG lines {.tabset}

Comparing gene lists for a given treatment comparison between all 3 ORG lines to identify commonalities in response between organoids.

#### nc(PAC - DMSO)
```{r}
### Venn Diagram df
x <- list(org38=rownames(org38_ncPAC_DEgenes),
          org49=rownames(org49_ncPAC_DEgenes),
          org66=rownames(org66_ncPAC_DEgenes))

x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`org38:org66`
```

0 genes were shared across all ORGs. 

`r length(common_genes)` were found to be common between org38 and org66:

  • `r common_genes`

#### nc(SN38 - DMSO)
```{r}
### Venn Diagram df
x <- list(org38=rownames(org38_ncSN38_DEgenes),
          org49=rownames(org49_ncSN38_DEgenes),
          org66=rownames(org66_ncSN38_DEgenes))

x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`org38:org49:org66`
```

`r length(common_genes)` genes were shared across all ORGs. 

  • `r common_genes`
  
#### pa(PAC - DMSO)
```{r}
### Venn Diagram df
x <- list(org38=rownames(org38_paPAC_DEgenes),
          org49=rownames(org49_paPAC_DEgenes),
          org66=rownames(org66_paPAC_DEgenes))

x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`org38:org66`
```

0 genes were shared across all ORGs. 

`r length(common_genes)` were found to be common between org38 and org66:

  • `r common_genes`
  
#### sn(SN38 - DMSO)
```{r}
### Venn Diagram df
x <- list(org38=rownames(org38_snSN38_DEgenes),
          org49=rownames(org49_snSN38_DEgenes),
          org66=rownames(org66_snSN38_DEgenes))

x.table <- venn(x, show.plot = FALSE)
x_attributes <- attributes(x.table)

ggvenn(x, 
       show_percentage = FALSE,
       text_size = 8)

common_genes <- x_attributes$intersections$`org38:org49:org66`
common_49_66 <- x_attributes$intersections$`org49:org66`
```

`r length(common_genes)` genes were shared across all ORGs. 

  • `r common_genes`
  
An additional `r length(common_49_66)` genes were found to be shared between ORG49 and ORG66:

  • `r common_49_66`

### Barcode Plots - P2 {.tabset}

#### ORG38 PAC
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncPAC_fit$t[,1]

### second (comparison set)
set2_upregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC < 0,])

### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


#### ORG38 SN38
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncSN38_fit$t[,1]

### second (comparison set)
set2_upregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC < 0,])

### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG49 SN38
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org49_ncSN38_fit$t[,1]

### second (comparison set)
set2_upregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC < 0,])

### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG66 PAC
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_ncPAC_fit$t[,1]

### second (comparison set)
set2_upregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC < 0,])

### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

```

#### ORG66 SN38
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_ncSN38_fit$t[,1]

### second (comparison set)
set2_upregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC < 0,])

### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### NC PAC VS SN38

ORG38: 
```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in ncSN38 - ncDMSO","downregulated in ncSN38 - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org38_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ncPAC - ncDMSO","downregulated in ncPAC - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```



ORG66:

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_ncPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in ncSN38 - ncDMSO","downregulated in ncSN38 - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ncPAC - ncDMSO","downregulated in ncPAC - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### paPAC vs snSN38

ORG38: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_paPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in paPAC-paDMSO") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org38_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in snSN38-snDMSO") 
legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


ORG66:

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_paPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in paPAC-paDMSO") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in snSN38-snDMSO") 
legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG38 vs ORG66 (NC)

NC_PAC - NC_DMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in ORG66 ncPAC-ncDMSO","downregulated in ORG66 ncPAC-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_ncPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 ncPAC-ncDMSO") 
legend("top",legend=c("upregulated in ORG38 ncPAC-ncDMSO","downregulated in ORG38 ncPAC-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

NC_SN38 - NC_DMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG66 ncSN38-ncDMSO","downregulated in ORG66 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG38 ncSN38-ncDMSO","downregulated in ORG38 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG38 vs ORG49 (NC)

NC_SN38 - NC_DMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG49 ncSN38-ncDMSO","downregulated in ORG49 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org49_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG49 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG38 ncSN38-ncDMSO","downregulated in ORG38 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


#### ORG49 vs ORG66 (NC)

NC_SN38 - NC_DMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG49 ncSN38-ncDMSO","downregulated in ORG49 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org49_ncSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG49 ncSN38-ncDMSO") 
legend("top",legend=c("upregulated in ORG66 ncSN38-ncDMSO","downregulated in ORG66 ncSN38-ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


#### ORG38 vs ORG66 (pre-treated)

paPAC - paDMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_paPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 paPAC-paDMSO") 
legend("top",legend=c("upregulated in ORG66 paPAC-paDMSO","downregulated in ORG66 paPAC-paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_paPAC_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 paPAC-paDMSO") 
legend("top",legend=c("upregulated in ORG38 paPAC-paDMSO","downregulated in ORG38 paPAC-paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


snSN38 - snDMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG66 snSN38-snDMSO","downregulated in ORG66 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org66_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG38 snSN38-snDMSO","downregulated in ORG38 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG38 vs ORG49 (pre-treated)

snSN38 - snDMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org38_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG38 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG49 snSN38-snDMSO","downregulated in ORG49 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org49_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG49 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG38 snSN38-snDMSO","downregulated in ORG38 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

#### ORG49 vs ORG66 (pre-treated)

snSN38 - snDMSO: 

```{r, fig.height=5, fig.width=6}
### first set 
statistic <- org66_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG66 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG49 snSN38-snDMSO","downregulated in ORG49 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### first set 
statistic <- org49_snSN38_fit$t[,1]
### second (comparison set)
set2_upregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC > 0,])
set2_downregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC < 0,])
### upregulated barcode plot
limma::barcodeplot(statistic = statistic,index=set2_upregulated,index2=set2_downregulated,xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "Genes ranked according to t-statistic in ORG49 snSN38-snDMSO") 
legend("top",legend=c("upregulated in ORG66 snSN38-snDMSO","downregulated in ORG66 snSN38-snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = statistic,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

### Barcode Plots - P1 VS P2

```{r, fig.height=5, fig.width=6}
### first set 
org38_P1_pac_fit <- readRDS('output/P1/org38_pac_fit.rds') 
t_stat <- org38_P1_pac_fit$t[,1]

### compared to ncPAC-ncDMSO
### second (comparison set)
set2_upregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org38_ncPAC_DEgenes[org38_ncPAC_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

### upregulated barcode plot
limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to PAC-DMSO in P1") 

legend("top",legend=c("upregulated in ncPAC - ncDMSO","downregulated in ncPAC - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### compared to paPAC-paDMSO
### second (comparison set)
set2_upregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org38_paPAC_DEgenes[org38_paPAC_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

### upregulated barcode plot
limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to PAC-DMSO in P1") 

legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

```{r, fig.height=5, fig.width=6}
### first set 
org38_P1_sn38_fit <- readRDS('output/P1/org38_sn38_fit.rds') 
t_stat <- org38_P1_sn38_fit$t[,1]

### compared to ncSN38-ncDMSO
### second (comparison set)
set2_upregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org38_ncSN38_DEgenes[org38_ncSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

### upregulated barcode plot
limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 

legend("top",legend=c("upregulated in ncSN38 - ncDMSO","downregulated in ncSN38 - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### compared to snSN38-snDMSO
### second (comparison set)
set2_upregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org38_snSN38_DEgenes[org38_snSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

### upregulated barcode plot
limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 

legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

```{r, fig.height=5, fig.width=6}
### first set 
org49_P1_sn38_fit <- readRDS('output/P1/org49_sn38_fit.rds') 
t_stat <- org49_P1_sn38_fit$t[,1]

### compared to ncSN38- ncDMSO
### second (comparison set)
set2_upregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org49_ncSN38_DEgenes[org49_ncSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 
legend("top",legend=c("upregulated in ncSN38 - ncDMSO","downregulated in ncSN38 - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### compared to snSN38-snDMSO
### second (comparison set)
set2_upregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org49_snSN38_DEgenes[org49_snSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```

```{r, fig.height=5, fig.width=6}
### first set 
org66_P1_pac_fit <- readRDS('output/P1/org66_pac_fit.rds') 
t_stat <- org66_P1_pac_fit$t[,1]

### compared to ncPAC-ncDMSO
### second (comparison set)
set2_upregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org66_ncPAC_DEgenes[org66_ncPAC_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to PAC-DMSO in P1") 
legend("top",legend=c("upregulated in ncPAC - ncDMSO","downregulated in ncPAC - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### compared to paPAC-paDMSO
### second (comparison set)
set2_upregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org66_paPAC_DEgenes[org66_paPAC_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to PAC-DMSO in P1") 
legend("top",legend=c("upregulated in paPAC - paDMSO","downregulated in paPAC - paDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


```{r, fig.height=5, fig.width=6}
### first set 
org66_P1_sn38_fit <- readRDS('output/P1/org66_sn38_fit.rds') 
t_stat <- org66_P1_sn38_fit$t[,1]

### compared to ncSN38-ncDMSO
### second (comparison set)
set2_upregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org66_ncSN38_DEgenes[org66_ncSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 
legend("top",legend=c("upregulated in ncSN38 - ncDMSO","downregulated in ncSN38 - ncDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)

### compared to snSN38-snDMSO
### second (comparison set)
set2_upregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC > 0,])
set2_upregulated<- names(t_stat[set2_upregulated])
set2_upregulated <- set2_upregulated[!is.na(set2_upregulated)]

set2_downregulated <- rownames(org66_snSN38_DEgenes[org66_snSN38_DEgenes$logFC < 0,])
set2_downregulated<- names(t_stat[set2_downregulated])
set2_downregulated <- set2_downregulated[!is.na(set2_downregulated)]

limma::barcodeplot(statistic=t_stat,index=set2_upregulated, index2=set2_downregulated, xlab='t-statistic',labels=c("downregulated", "upregulated"), main= "P2 ranked according to SN38-DMSO in P1") 
legend("top",legend=c("upregulated in snSN38 - snDMSO","downregulated in snSN38 - snDMSO"),lwd=2,box.lty=0,col=c("red","blue"), cex=0.8, border=NULL, inset=c(0.3,-0.1), xpd=NA)

### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = t_stat,
       index = list(upregulated=set2_upregulated, downregulated=set2_downregulated),
       use.ranks = TRUE)
```


## Predefined Geneset Testing 

### Hallmark Genesets {.tabset}
```{r, warning=FALSE}
### predefined gene sets 
msigdb_hallmark_dnarepair <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_DNA_REPAIR.v2025.1.Hs.gmt')
msigdb_hallmark_apoptosis <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_APOPTOSIS.v2025.1.Hs.gmt')
msigdb_hallmark_EMT <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.v2025.1.Hs.gmt')
msigdb_hallmark_p53 <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_P53_PATHWAY.v2025.1.Hs.gmt')
msigdb_hallmark_G2M <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_G2M_CHECKPOINT.v2025.1.Hs.gmt')
msigdb_hallmark_Inflammatory_response <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_INFLAMMATORY_RESPONSE.v2025.1.Hs.gmt')
msigdb_hallmark_Mitotic_Spindle <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_MITOTIC_SPINDLE.v2025.1.Hs.gmt')
msigdb_hallmark_Notch_signalling <-read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_NOTCH_SIGNALING.v2025.1.Hs.gmt')
msigdb_hallmark_Oxidative <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_OXIDATIVE_PHOSPHORYLATION.v2025.1.Hs.gmt')
msigdb_hallmark_WNT_BETA_CATENIN <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_WNT_BETA_CATENIN_SIGNALING.v2025.1.Hs.gmt')
msigdb_hallmark_E2F <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_E2F_TARGETS.v2025.1.Hs.gmt')
msigdb_hallmark_COMPLEMENT <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_COMPLEMENT.v2025.1.Hs.gmt')
msigdb_hallmark_KRAS_UP <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_KRAS_SIGNALING_UP.v2025.1.Hs.gmt')
msigdb_hallmark_KRAS_DN <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_KRAS_SIGNALING_DN.v2025.1.Hs.gmt')
msigdb_hallmark_INTERFERON_GAMMA <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_INTERFERON_GAMMA_RESPONSE.v2025.1.Hs.gmt')
msigdb_hallmark_INTERFERON_ALPHA <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_INTERFERON_ALPHA_RESPONSE.v2025.1.Hs.gmt')
msigdb_hallmark_IL6_JAKSTAT3 <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_IL6_JAK_STAT3_SIGNALING.v2025.1.Hs.gmt')
msigdb_hallmark_IL2_STAT5 <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_IL2_STAT5_SIGNALING.v2025.1.Hs.gmt')
msigdb_hallmark_ROS <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.v2025.1.Hs.gmt')
msigdb_hallmark_MTORC1 <- read.gmt('data/MSigDB_Hallmark_Genesets/HALLMARK_MTORC1_SIGNALING.v2025.1.Hs.gmt')

msigdb_hallmark_geneset_list <- list(DNA_repair = msigdb_hallmark_dnarepair$gene, 
                    APOPTOSIS = msigdb_hallmark_apoptosis$gene,
                    EMT = msigdb_hallmark_EMT$gene,
                    p53_PATHWAY = msigdb_hallmark_p53$gene,
                    G2M_CHECKPOINT = msigdb_hallmark_G2M$gene,
                    INFLAMMATORY_RESPONSE = msigdb_hallmark_Inflammatory_response$gene,
                    MITOTIC_SPINDLE = msigdb_hallmark_Mitotic_Spindle$gene,
                    NOTCH_SIGNALLING = msigdb_hallmark_Notch_signalling$gene,
                    OXIDATIVE_PHOSPHORYLATION = msigdb_hallmark_Oxidative$gene,
                    WNT_BETA_CATENIN_SIGNALLING = msigdb_hallmark_WNT_BETA_CATENIN$gene,
                    E2F = msigdb_hallmark_E2F$gene,
                    COMPLEMENT = msigdb_hallmark_COMPLEMENT$gene,
                    KRAS_UP = msigdb_hallmark_KRAS_UP$gene,
                    KRAS_DN = msigdb_hallmark_KRAS_DN$gene,
                    INTERFERON_GAMMA = msigdb_hallmark_INTERFERON_GAMMA$gene,
                    INTERFERON_ALPHA = msigdb_hallmark_INTERFERON_ALPHA$gene,
                    IL6_JAKSTAT3 = msigdb_hallmark_IL6_JAKSTAT3$gene,
                    IL2_STAT5 = msigdb_hallmark_IL2_STAT5$gene,
                    ROS = msigdb_hallmark_ROS$gene,
                    MTORC1_SIGNALLING = msigdb_hallmark_MTORC1$gene)
```

#### ORG38 

```{r}
### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = org38_ncPAC_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_paPAC_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_ncSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_snSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)
```

#### ORG49 

```{r}
### competitive gene set testing (pre-ranked)
# limma::cameraPR(statistic = org49_ncPAC_fit$t[,1],
#        index = list(DNA_repair = msigdb_hallmark_dnarepair$gene, 
#                     APOPTOSIS = msigdb_hallmark_apoptosis$gene,
#                     EMT = msigdb_hallmark_EMT$gene,
#                     p53_PATHWAY = msigdb_hallmark_p53$gene),
#        use.ranks = TRUE)
# 
# limma::cameraPR(statistic = org49_paPAC_fit$t[,1],
#        index = list(DNA_repair = msigdb_hallmark_dnarepair$gene, 
#                     APOPTOSIS = msigdb_hallmark_apoptosis$gene,
#                     EMT = msigdb_hallmark_EMT$gene,
#                     p53_PATHWAY = msigdb_hallmark_p53$gene),
#        use.ranks = TRUE)

limma::cameraPR(statistic = org49_ncSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org49_snSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)
```

#### ORG66 

```{r}
### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = org66_ncPAC_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org66_paPAC_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org66_ncSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org66_snSN38_fit$t[,1],
       index = msigdb_hallmark_geneset_list,
       use.ranks = TRUE)
```
### Cisplatin resistance {.tabset}

```{r}
### predefined gene sets 
msigdb_LI_cisplatin_UP <- read.gmt('data/MSigDB_Curated_Genesets/LI_CISPLATIN_RESISTANCE_UP.v2025.1.Hs.gmt')
msigdb_LI_cisplatin_DN <- read.gmt('data/MSigDB_Curated_Genesets/LI_CISPLATIN_RESISTANCE_DN.v2025.1.Hs.gmt')
msigdb_TSUNODA_cisplatin_UP <- read.gmt('data/MSigDB_Curated_Genesets/TSUNODA_CISPLATIN_RESISTANCE_UP.v2025.1.Hs.gmt')
msigdb_TSUNODA_cisplatin_DN <- read.gmt('data/MSigDB_Curated_Genesets/TSUNODA_CISPLATIN_RESISTANCE_DN.v2025.1.Hs.gmt')
msigdb_WHITESIDE_cisplatin_UP <- read.gmt('data/MSigDB_Curated_Genesets/WHITESIDE_CISPLATIN_RESISTANCE_UP.v2025.1.Hs.gmt')
msigdb_WHITESIDE_cisplatin_DN <- read.gmt('data/MSigDB_Curated_Genesets/WHITESIDE_CISPLATIN_RESISTANCE_DN.v2025.1.Hs.gmt')

msigdb_curated_list <- list(LI_UP = msigdb_LI_cisplatin_UP$gene, 
                    LI_DN = msigdb_LI_cisplatin_DN$gene,
                    TSUNODA_UP = msigdb_TSUNODA_cisplatin_UP$gene,
                    TSUNODA_DN = msigdb_TSUNODA_cisplatin_DN$gene,
                    WHITESIDE_UP = msigdb_WHITESIDE_cisplatin_UP$gene,
                    WHITESIDE_DN = msigdb_WHITESIDE_cisplatin_DN$gene )
```

#### ORG38 

```{r}
### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = org38_ncPAC_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_paPAC_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_ncSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org38_snSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)
```
#### ORG49 

```{r}
### competitive gene set testing (pre-ranked)
# limma::cameraPR(statistic = org49_ncPAC_fit$t[,1],
#        index = msigdb_curated_list,
#        use.ranks = TRUE)

# limma::cameraPR(statistic = org49_paPAC_fit$t[,1],
#        index = msigdb_curated_list,
#        use.ranks = TRUE)

limma::cameraPR(statistic = org49_ncSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)

limma::cameraPR(statistic = org49_snSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)
```

#### ORG66 

```{r}
### competitive gene set testing (pre-ranked)
limma::cameraPR(statistic = org66_ncPAC_fit$t[,1],
        index = msigdb_curated_list,
        use.ranks = TRUE)

limma::cameraPR(statistic = org66_paPAC_fit$t[,1],
        index = msigdb_curated_list,
        use.ranks = TRUE)

limma::cameraPR(statistic = org66_ncSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)

a <- limma::cameraPR(statistic = org66_snSN38_fit$t[,1],
       index = msigdb_curated_list,
       use.ranks = TRUE)
```
