---
title: "Pathway Enrichment Visualisation"
output: html_document
date: "2024-08-28"
---

```{r setup}
knitr::opts_chunk$set(
	echo = FALSE,
	message = FALSE,
	warning = FALSE
)
library(tidyverse)
library(dplyr)
library(DOSE)
library(enrichplot)
library(clusterProfiler)
library(org.Hs.eg.db)
library(ggpubr)
```

```{r}

setwd('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/')

#set up and load saved gse objects

ORG38_PAC50_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG38/PAC_50_48H_gse.rds')
ORG38_SN3850_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG38/SN38_50_48H_gse.rds')

ORG49_PAC50_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG49/PAC_50_48H_gse.rds')
ORG49_SN3850_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG49/SN38_50_48H_gse.rds')

ORG66_PAC50_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG66/PAC_50_48H_gse.rds')
ORG66_SN3850_48H_gse <- readRDS('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG66/SN38_50_48H_gse.rds')

#set up and load DE gene dataframes
ORG38_PAC50_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/PAC_IC50_48H.txt', show_col_types = FALSE)
ORG38_SN3850_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/SN38_IC50_48H.txt', show_col_types = FALSE)

ORG49_PAC50_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/PAC_IC50_48H.txt', show_col_types = FALSE)
ORG49_SN3850_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/SN38_IC50_48H.txt', show_col_types = FALSE)

ORG66_PAC50_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/PAC_IC50_48H.txt', show_col_types = FALSE)
ORG66_SN3850_48H_DE <- read_delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/SN38_IC50_48H.txt', show_col_types = FALSE)
```

## Threshold notes
• Differentially expressed genes = meeting logFC threshold of x > +1 and/or x < -1

• FDR threshold for differential expression analysis = 0.005

• p value threshold for GSEA analysis = 0.005



## ORG 38 
### 48H timepoint comparisons
```{r, label=org38_48H_plots, fig.height=10, fig,width=25}
#ORG38 Paclitaxel plot summary 
ORG38_PAC50_48H_DE <- as.data.frame(ORG38_PAC50_48H_DE)

fc <- ORG38_PAC50_48H_DE[order(ORG38_PAC50_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p1 <- cnetplot(ORG38_PAC50_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=c('RNA binding','organelle envelope','mitochondrial protein-containing complex','chromosome organization','DNA metabolic process', 'cell cycle process'),
         colorEdge=TRUE,
         cex.params = list(category_label=1),
         color.params=list(category='black')) + ggtitle('ORG38 Paclitaxel IC50')

p1 + scale_color_gradientn(colours = c("red3", "blue")) 


ORG38_SN3850_48H_DE <- as.data.frame(ORG38_SN3850_48H_DE)
fc <- ORG38_SN3850_48H_DE[order(ORG38_SN3850_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p2 <- cnetplot(ORG38_SN3850_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=c('organic cyclic compound binding','organelle lumen','protein-containing complex','phosphorus metabolic process','cellular nitrogen compound metabolic process','RNA binding'),
         colorEdge=TRUE,
         cex.params = list(category_label=1.5),
         color.params=list(category='black'))+ ggtitle('ORG38 SN38 IC50')

p2 + scale_color_gradientn(colours = c("red3", "blue"))

### Dotplots
```{r, fig.height=8, fig,width=3}
dotplot(ORG38_PAC50_48H_gse,showCategory=20) + ggtitle("ORG38, Paclitaxel IC50 dotplot")
dotplot(ORG38_SN3850_48H_gse,showCategory=20) + ggtitle("ORG38, SN38 IC50 dotplot")
```

## ORG 49
### 48H timepoint comparisons
```{r, label=org49_48H_plots, fig.height=10, fig,width=25}

#ORG49 Paclitaxel plot summary 
ORG49_PAC50_48H_DE <- as.data.frame(ORG49_PAC50_48H_DE)

fc <- ORG49_PAC50_48H_DE[order(ORG49_PAC50_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p1 <- cnetplot(ORG49_PAC50_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=6,
         colorEdge=TRUE,
         cex.params = list(category_label=1.5),
         color.params=list(category='black'))
p1 + scale_color_gradientn(colours = c("red3", "blue")) + ggtitle('ORG49 Paclitaxel IC50')


ORG49_SN3850_48H_DE <- as.data.frame(ORG49_SN3850_48H_DE)

fc <- ORG49_SN3850_48H_DE[order(ORG49_SN3850_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p2 <- cnetplot(ORG49_SN3850_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=c('intracellular anatomical structure','small molecule metabolic process','primary metabolic process','steroid metabolic process','catalytic activity','cholesterol metabolic process'),
         colorEdge=TRUE,
         cex.params = list(category_label=1.5),
         color.params=list(category='black'))+ ggtitle('ORG49 SN38 IC50')

p2 + scale_color_gradientn(colours = c("red3", "blue"))
```

### Dotplots
```{r, label=dotplots, fig.height=8, fig,width=3}
dotplot(ORG49_PAC50_48H_gse,showCategory=20) + ggtitle("ORG49, Paclitaxel IC50 dotplot")
dotplot(ORG49_SN3850_48H_gse,showCategory=20) + ggtitle("ORG49, SN38 IC50 dotplot")
```

## ORG 66
### 48H timepoint comparisons
```{r, label=org66_48H_plots, fig.height=12, fig,width=25}

#ORG66 Paclitaxel plot summary 
ORG66_PAC50_48H_DE <- as.data.frame(ORG66_PAC50_48H_DE)
fc <- ORG66_PAC50_48H_DE[order(ORG66_PAC50_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p1 <- cnetplot(ORG66_PAC50_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=6,
         colorEdge=TRUE,
         cex.params = list(category_label=1.5),
         color.params=list(category='black')) + ggtitle('ORG66 Paclitaxel IC50')

p1 + scale_color_gradientn(colours = c("red3", "blue")) 


ORG66_SN3850_48H_DE <- as.data.frame(ORG66_SN3850_48H_DE)
fc <- ORG66_SN3850_48H_DE[order(ORG66_SN3850_48H_DE$logFC, decreasing = TRUE),]
gene_names <- fc$genes
fc<- fc[,c("logFC")]
names(fc) <- gene_names

p2 <- cnetplot(ORG66_SN3850_48H_gse, 
         foldChange=fc,
         node_label='gene',
         showCategory=c('cell division','mitotic cell cycle','chromosome segregation','spindle assembly','organelle fission','cell cycle process'),
         colorEdge=TRUE,
         cex.params = list(category_label=1.5),
         color.params=list(category='black')) + ggtitle('ORG66 SN38 IC50')

p2 + scale_color_gradientn(colours = c("red3", "blue"))

```
### Dotplots
```{r, label= dotplots66, fig.height=8, fig,width=3}
dotplot(ORG66_PAC50_48H_gse,showCategory=20) + ggtitle("ORG66, Paclitaxel IC50 dotplot")
dotplot(ORG66_SN3850_48H_gse,showCategory=20) + ggtitle("ORG66, SN38 IC50 dotplot")
```
