---
title: "Differential Gene Expression 48H"
output: html_document
date: "2024-08-15"
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
library(Seurat)
library(gtools)
library(tidyseurat)
library(DESeq2)
library(edgeR)
library(RUVSeq)
library(variancePartition)
library(reshape2)
library(gridExtra)
library(ggrepel)
library(fgsea)
library(org.Hs.eg.db)
library(DT)
library(DOSE)
library(enrichplot)
library(clusterProfiler)
library(dplyr)
library(pheatmap)
library(EnhancedVolcano)
library(msigdbr)
```

```{r, warning=FALSE}
#import metadata
metadata<-read.csv("/Volumes/bioinf/home/ndalvi/MACseq_PMC228/metadata/PMC228.csv",header=T) %>%
  arrange("Barcode") %>%
  mutate(id=gsub("Plate","",Plate_ID)) %>%
  mutate(label=paste0(Treatment_1,
                      " ","(",Well_ID,")")) %>%
  mutate(analysis_condition=paste0(Organoid,".",Treatment_1,".",Time))

#load in count data 
raw_counts <- Read10X(data.dir = '/Users/dalvinikita/Documents/MACseq_PMC228/raw/PMC228/starsolo/SA_PMC228-null.Solo.out/Gene/raw')

#filter by cpm>10 in at least 2 samples
keep <- rowSums(edgeR::cpm(raw_counts)>10) >= 2
raw_counts <- raw_counts[keep,]

#filter for the minimum number of reads per gene per sample, as per edgeR
row.names(metadata)<-metadata$label

mac <- CreateSeuratObject(counts = raw_counts,
                          project = 'PMC228',
                          min.cells = 1,
                          min.features = 1)

mac[["percent.mt"]]<-PercentageFeatureSet(mac, pattern = "^mt-")
mac[["percent.rp"]]<-PercentageFeatureSet(mac, pattern = "^rp-")

#add location metadata
mac<- mac %>%
  inner_join(metadata,by=c(".cell"="Barcode"))

#exclude mito and ribo genes from the counts
mac<-mac[!grepl("^MT-",rownames(mac))]
mac<-mac[!grepl("^RP[SL][[:digit:]]|^RPLP[[:digit:]]|^RPSA",rownames(mac))]

############# Differential expression
plot_volcano<-function(top, FDR_cutoff=0.005){
  top<-as.data.frame(top)
  top$diffexpressed<-"NO"
  if(any(colnames(top) %in% "adj.P.Val")){
    top<-top %>%
      rename('FDR'='adj.P.Val')
  }
  top[top$logFC>=1 & top$FDR<=FDR_cutoff,]$diffexpressed<-"UP"
  top[top$logFC<=-1 & top$FDR<=FDR_cutoff,]$diffexpressed<-"DOWN"

  top$labelgenes<-""
  top[top$diffexpressed!="NO",]$labelgenes<-rownames(top[top$diffexpressed!="NO",])

  ggplot(top, aes(x=logFC, y=-log10(FDR), col=diffexpressed, label=labelgenes))+
    geom_point()+
    theme_classic()+
    geom_text_repel(min.segment.length=5)+
    scale_color_manual(values=c("blue", "black", "red"))+
    geom_vline(xintercept=c(-1, 1), col="red") +
    geom_hline(yintercept=-log10(0.05), col="red")
}

diff_exprs_voomLmFit<-function(treatment_1,treatment_2,max.overlaps=30){
  countData<-mac@assays$RNA$counts[,grepl(paste0(treatment_1,"|",treatment_2),coldata$condition)]
  coldata_temp<-coldata[grepl(paste0(treatment_1,"|",treatment_2),coldata$condition),]
  group = coldata_temp$condition
# batch = coldata_temp$batch
  design = model.matrix(~0+group)
  edgeR<-DGEList(
    counts = countData+1,
    samples = coldata_temp$condition,
    group = coldata_temp$condition
  )
  edgeR <- calcNormFactors(edgeR,methods="TMMwsp")
  edgeR <- estimateDisp(edgeR,design)
  cpm <- edgeR::cpm(edgeR,log=T)
  #fit <- glmQLFit(edgeR, design)
  fit <- edgeR::voomLmFit(edgeR, design)
  myargs = list(paste0("group",treatment_1,"-",paste0("group",treatment_2)), levels=design)
  contrasts <- do.call(makeContrasts, myargs)
  tmp <- contrasts.fit(fit, contrasts)
  tmp <- eBayes(tmp)
  top.table <- topTable(tmp, sort.by = "P", n = Inf)

  return(as.data.frame(top.table))
}

diff_exprs_edgeR<-function(treatment_1,treatment_2){
  countData<-mac@assays$RNA$counts[,grepl(paste0(treatment_1,"|",treatment_2),coldata$condition)]
  coldata_temp<-coldata[grepl(paste0(treatment_1,"|",treatment_2),coldata$condition),]
  group = coldata_temp$condition
  batch = coldata_temp$batch
  design = model.matrix(~0+group+batch)
  edgeR<-DGEList(
    counts = countData+1,
    samples = coldata_temp$condition,
    group = coldata_temp$condition
  )
  edgeR <- calcNormFactors(edgeR,methods="TMMwsp")
  edgeR <- estimateDisp(edgeR,design)
  fit <- glmQLFit(edgeR, design)
  myargs = list(paste0("group",treatment_1,"-",paste0("group",treatment_2)), levels=design)
  contrasts <- do.call(makeContrasts, myargs)
  qlf <- glmQLFTest(fit, contrast=contrasts)
  topTags <- topTags(qlf,n=length(qlf$df.total))
  return(as.data.frame(topTags))
}

FDR_cutoff <- 0.005

treatment <- c("Paclitaxel IC12.5","Paclitaxel IC25", "Paclitaxel IC50","SN38 IC12.5","SN38 IC25","SN38 IC50")
ORG38_DEGs <- c(4974,5005,4810,828,1213,1213)
ORG49_DEGs <- c(7,27,108,4,173,472)
ORG66_DEGs <- c(1222,1460,1137,55,351,336)

DGE_summary <- data_frame(treatment, ORG38_DEGs, ORG49_DEGs, ORG66_DEGs)
```

### Summary of results
Summary of differential gene expression analysis for all organoids and treatments:

```{r}
DGE_summary
```

## Differential Expression Analysis
### ORG38 {.tabset}
#### Paclitaxel IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.PAC_IC12.5.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG38 Paclitaxel IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]
```

In summary: 4760 genes differentially expressed between Paclitaxel IC12.5 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### Paclitaxel IC25.0 48H -DMSO 48H
```{r, fig.height=7, fig.width=7,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.PAC_IC25.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)


EnhancedVolcano(DEgenes,
                lab = rownames(DEgenes),
                x = 'logFC',
                y = 'adj.P.Val')
  
plot_volcano(DEgenes)  + ggtitle("ORG38 Paclitaxel IC25 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
```


In summary: 4757 genes differentially expressed between Paclitaxel IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### Paclitaxel IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.PAC_IC50.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG38 Paclitaxel IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/PAC_IC50_48H.txt', sep='\t')
```

In summary: 4583 genes differentially expressed between Paclitaxel IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
# ##enrichment analysis
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 600,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# 
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG38/PAC_50_48H_gse.rds')
# 
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG38/PAC_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG38/PAC_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])

```

#### SN38 IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.SN38_IC12.5.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG38 SN38 IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

```

In summary: 785 genes differentially expressed between SN38 IC12.5 and DMSO at 48H:
```{r}
datatable(DEgenes)
```


#### SN38 IC25.0 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.SN38_IC25.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG38 SN38 IC25 VS DMSO")
DEgenes<- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)

```

In summary: 1188 genes differentially expressed between SN38 IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```


#### SN38 IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG38.SN38_IC50.48H"
treatment_2<-"ORG38.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG38 SN38 IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/SN38_IC50_48H.txt', sep='\t')
```

In summary: 1224 genes differentially expressed between SN38 IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
##enrichment analysis
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 600,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG38/SN38_50_48H_gse.rds')
# 
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG38/SN38_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG38/SN38_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])

```

### {-}


### ORG49 {.tabset}

#### Paclitaxel IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.PAC_IC12.5.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
#plot_volcano(DEgenes)  + ggtitle("ORG49 Paclitaxel IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]
```

In summary: 0 genes differentially expressed between Paclitaxel IC12.5 and DMSO at 48H

#### Paclitaxel IC25.0 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.PAC_IC25.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
#plot_volcano(DEgenes)  + ggtitle("ORG49 Paclitaxel IC25 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
```

In summary: 4 genes differentially expressed between Paclitaxel IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```


#### Paclitaxel IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.PAC_IC50.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG49 Paclitaxel IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/PAC_IC50_48H.txt', sep='\t')
```

In summary: 109 genes differentially expressed between Paclitaxel IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
# #enrichment analysis
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 600,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG49/PAC_50_48H_gse.rds')
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG49/PAC_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG49/PAC_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])
```

#### SN38 IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.SN38_IC12.5.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
#plot_volcano(DEgenes)  + ggtitle("ORG49 SN38 IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

```

In summary: 0 genes differentially expressed between SN38 IC12.5 and DMSO at 48H

#### SN38 IC25 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.SN38_IC25.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG49 SN38 IC25 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
```

In summary: 149 genes differentially expressed between SN38 IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### SN38 IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG49.SN38_IC50.48H"
treatment_2<-"ORG49.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG49 SN38 IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/SN38_IC50_48H.txt', sep='\t')
```

In summary: 454 genes differentially expressed between SN38 IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
#enrichment analysis
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 600,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# 
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG49/SN38_50_48H_gse.rds')
# 
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG49/SN38_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG49/SN38_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])
```

### {-}

### ORG66 {.tabset}

#### Paclitaxel IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.PAC_IC12.5.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 Paclitaxel IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]
```

In summary: 381 genes differentially expressed between Paclitaxel IC12.5 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### Paclitaxel IC25.0 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.PAC_IC25.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 Paclitaxel IC25 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
```

In summary: 1181 genes differentially expressed between Paclitaxel IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### Paclitaxel IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.PAC_IC50.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 Paclitaxel IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/PAC_IC50_48H.txt', sep='\t')
```

In summary: 862 genes differentially expressed between Paclitaxel IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
#enrichment analysis#
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 400,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG66/PAC_50_48H_gse.rds')
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG66/PAC_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG66/PAC_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])
```

#### SN38 IC12.5 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.SN38_IC12.5.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 SN38 IC12.5 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]
```

In summary: 14 genes differentially expressed between SN38 IC12.5 and DMSO at 48H:
```{r}
datatable(DEgenes)
```


#### SN38 IC25.0 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.SN38_IC25.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 SN38 IC25 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]
DEgenes['genes']<- rownames(DEgenes)
```

In summary: 43 genes differentially expressed between SN38 IC25 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

#### SN38 IC50 48H -DMSO 48H
```{r, fig.height=10, fig.width=10,warning=FALSE, message=FALSE, echo=FALSE}
treatment_1<-"ORG66.SN38_IC50.48H"
treatment_2<-"ORG66.DMSO.48H"
coldata<-data.frame(batch=rep(1,384),condition=mac$analysis_condition)
DEgenes<-diff_exprs_voomLmFit(treatment_1,treatment_2)
plot_volcano(DEgenes)  + ggtitle("ORG66 SN38 IC50 VS DMSO")
DEgenes <- DEgenes[DEgenes$logFC>=1 & DEgenes$adj.P.Val<=FDR_cutoff | DEgenes$logFC<=-1 & DEgenes$adj.P.Val<=FDR_cutoff ,]

DEgenes['genes']<- rownames(DEgenes)
write.table(DEgenes,row.names = F, quote=F, col.names=T, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/SN38_IC50_48H.txt', sep='\t')
```

In summary: 131 genes differentially expressed between SN38 IC50 and DMSO at 48H:
```{r}
datatable(DEgenes)
```

Enriched pathways for differentially expressed genes:
```{r}
# #enrichment analysis
# gene_list <- DEgenes[,c(1)]
# names(gene_list) <- DEgenes$genes
# gene_list = sort(gene_list, decreasing = TRUE)
# 
# gse <- gseGO(geneList=gene_list,
#              ont ="ALL",
#              keyType = "ALIAS",
#              minGSSize = 3,
#              maxGSSize = 400,
#              pvalueCutoff = 0.005,
#              verbose = TRUE,
#              OrgDb = "org.Hs.eg.db",
#              pAdjustMethod = "none")
# saveRDS(gse, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/rObjects/ORG66/SN38_50_48H_gse.rds')
# output <- as.data.frame(gse@result)
# output <- output[order(output$pvalue),]
# rownames(output)<- NULL
# write.table(output, row.names = FALSE, quote=FALSE, col.names = TRUE, file='/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG66/SN38_IC50_48H.txt', sep='\t')

output <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/enrichedPathways/ORG66/SN38_IC50_48H.txt', header = TRUE)
datatable(output[,c(3,4,5,7,12)])
```

## Heatmaps and visualisation
### Resistance Genes {.tabset}

```{r}
#48H
orgs <- c('ORG38','ORG49','ORG66')

transport_genes <- c('ABCC1', 'ABCC2', 'ABCC3', 'ABCC4', 'ABCC5', 'ABCG2','SLCO1B3','RPS20', 'RPL26', 'RPL10A', 'RPL39', 'RPL7', 'RPL34', 'EIF4A2', 'EEF1D')

paclitaxel_resistance_genes<-c('MFAP2','ITGA2','PHTF2','CCL26','TSPAN9','MLLT11')

sn38_resistance_genes <- c('NDUFA2', 'SDHD', 'LSM5', 'DCAF4', 'COX10', 'RBM8A', 'TIMP1', 'QKI', 'TGOLN2','PTGS2')
```

#### Drug Transporters
Looking specifically at genes involved in resistance and drug transport (ABS, SLC family of drug transporters + Ribosomal genes)

• ABCC1, ABCC2, ABCC3, ABCC4, ABCC5, ABCG2

• SLC01B3 (many SLC genes removed in initial filtering due to low read counts)

• RPS20, RPL26, RPL10A, RPL39, RPL7, RPL34 

• EIF4A2, EEF1D

```{r}
# re-load count data because we want to keep all genes:
#load in count data 
raw_counts <- Read10X(data.dir = '/Users/dalvinikita/Documents/MACseq_PMC228/raw/PMC228/starsolo/SA_PMC228-null.Solo.out/Gene/raw')

#filter by cpm>10 in at least 2 samples
keep <- rowSums(edgeR::cpm(raw_counts)>10) >= 2
raw_counts <- raw_counts[keep,]

#filter for the minimum number of reads per gene per sample, as per edgeR
row.names(metadata)<-metadata$label

mac <- CreateSeuratObject(counts = raw_counts,
                          project = 'PMC228',
                          min.cells = 1,
                          min.features = 1)

mac[["percent.mt"]]<-PercentageFeatureSet(mac, pattern = "^mt-")
mac[["percent.rp"]]<-PercentageFeatureSet(mac, pattern = "^rp-")

#add location metadata
mac<- mac %>%
  inner_join(metadata,by=c(".cell"="Barcode"))

#Normalize to cpm and log transform counts 
data <- mac %>%
  NormalizeData(normalization.method = 'RC', scale.factor = 1e6) 
data@assays$RNA$counts <- log1p(data@assays$RNA$counts)
colnames(data)<- data$label


for (organoid in orgs) {
  counts <-  data@assays$RNA$counts[transport_genes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'Paclitaxel_IC50'|columns$Treatment_1== 'SN38_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  a <- pheatmap(counts)
  paste(a)
}
```

#### Paclitaxel Resistance
```{r}
for (organoid in orgs) {
  counts <-  data@assays$RNA$counts[paclitaxel_resistance_genes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  a <- pheatmap(counts)
  paste(a)
}
```

#### SN38 Resistance
```{r}
for (organoid in orgs) {
  counts <-  data@assays$RNA$counts[sn38_resistance_genes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  a <- pheatmap(counts)
  paste(a)
}
```
#### Housekeeping Genes 
```{r}
# load human gene sets from MSigDB
# select genes from "HOUSEKEEPING GENES" gene set only
cgp_gene_sets = msigdbr(species = "human", category = "C2", subcategory = "CGP")
housekeeping_genes <- cgp_gene_sets[cgp_gene_sets$gs_name=='HSIAO_HOUSEKEEPING_GENES',c('gene_symbol')]

# keep only genes in the geneset that also appear in filtered counts data
gene_names <- c(rownames(data@assays$RNA$counts))
housekeeping_genes <- subset(housekeeping_genes, gene_symbol %in% gene_names)
housekeeping_genes <- c(housekeeping_genes$gene_symbol)

#Creating heatmaps
for (organoid in orgs) {
  counts <-  data@assays$RNA$counts[housekeeping_genes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  new_metadata <-columns[columns$Organoid==paste0(organoid),]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  sample_annotation_df <- data.frame(SPLINTR = new_metadata$SPLINTR) 
  rownames(sample_annotation_df) <- new_metadata$label
  a <- pheatmap(counts,
                annotation_col = sample_annotation_df,
                main = paste('MSigDB human housekeeping genes in',organoid))
  paste(a)
}

### Using HKGs from Bulk RNA-seq
# load HKGs from bulk RNAseq
hkg = read.table('/Users/dalvinikita/Downloads/HK_genes.txt', header = FALSE)
housekeeping_genes <- data.frame(gene_symbol=hkg$V1)

# keep only genes in the geneset that also appear in filtered counts data
gene_names <- c(rownames(data@assays$RNA$counts))
housekeeping_genes <- subset(housekeeping_genes, gene_symbol %in% gene_names)
housekeeping_genes <- c(housekeeping_genes$gene_symbol)

#Creating heatmaps
for (organoid in orgs) {
  counts <-  data@assays$RNA$counts[housekeeping_genes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  new_metadata <-columns[columns$Organoid==paste0(organoid),]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  sample_annotation_df <- data.frame(SPLINTR = new_metadata$SPLINTR) 
  rownames(sample_annotation_df) <- new_metadata$label
  a <- pheatmap(counts,
                annotation_col = sample_annotation_df,
                main = paste('HKGs from RNAseq in',organoid))
  paste(a)
}
```
#### DE Genes 
```{r}
# load DE genes for Paclitaxel IC50 at 48H
org38_pacDE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/PAC_IC50_48H.txt', header = TRUE)
org49_pacDE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/PAC_IC50_48H.txt', header = TRUE)
org66_pacDE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/PAC_IC50_48H.txt', header = TRUE)
pac <- list(ORG38=org38_pacDE,ORG49=org49_pacDE,ORG66=org66_pacDE)

# load DE genes for SN38 IC50 at 48H
org38_sn38DE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG38/SN38_IC50_48H.txt', header = TRUE)
org49_sn38DE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG49/SN38_IC50_48H.txt', header = TRUE)
org66_sn38DE <- read.delim('/Users/dalvinikita/Documents/GitHub/MACseq-SPLINTR-Project/data/DEgenes/ORG66/SN38_IC50_48H.txt', header = TRUE)
sn38 <- list(ORG38=org38_sn38DE,ORG49=org49_sn38DE,ORG66=org66_sn38DE )

for (organoid in orgs) {
  DEgenes <- pac[paste(organoid)]
  DEgenes <- c(DEgenes[[1]]$genes)
  counts <-  data@assays$RNA$counts[DEgenes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  new_metadata <-columns[columns$Organoid==paste0(organoid),]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  sample_annotation_df <- data.frame(SPLINTR = new_metadata$SPLINTR) 
  rownames(sample_annotation_df) <- new_metadata$label
  a <- pheatmap(counts,
                annotation_col = sample_annotation_df,
                main = paste('Paclitaxel DE genes in',organoid),
                border_color = 'NA',
                cellwidth = 10)
  paste(a)
}

for (organoid in orgs) {
  DEgenes <- sn38[paste(organoid)]
  DEgenes <- c(DEgenes[[1]]$genes)
  counts <-  data@assays$RNA$counts[DEgenes,]
  columns <- metadata[metadata$Organoid==organoid &
                                metadata$Time=='48H',]
  new_metadata <-columns[columns$Organoid==paste0(organoid),]
  columns <- c(columns[columns$Treatment_1== 'DMSO' |columns$Treatment_1== 'SN38_IC50'|columns$Treatment_1== 'Paclitaxel_IC50', c("label")])
  counts <- counts[,columns] %>%
    as.data.frame() 
  counts <- counts[!is.infinite(rowSums(counts)),]
  sample_annotation_df <- data.frame(SPLINTR = new_metadata$SPLINTR) 
  rownames(sample_annotation_df) <- new_metadata$label
  a <- pheatmap(counts,
                annotation_col = sample_annotation_df,
                main = paste('SN38 DE genes in',organoid),
                border_color = 'NA',
                cellwidth = 10)
  paste(a)
}
```



